arcgis-ios-sdk-samples/Utility network/Create load report/CreateLoadReportViewController.swift

```diff
@@ -0,0 +1,268 @@
+            title: "Add a phase to get the load report.",
+            message: nil,
+            preferredStyle: .actionSheet
+        )
+        remainingPhases.forEach { phase in
+            alertController.addAction(UIAlertAction(title: phase.name, style: .default) { _ in
+                self.phaseSummaries.append((phase, nil))
+                self.tableView.insertRows(at: [IndexPath(row: self.phaseSummaries.endIndex - 1, section: 0)], with: .automatic)
+            })
+        }
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel)
+        alertController.addAction(cancelAction)
+        alertController.popoverPresentationController?.barButtonItem = sender
+        present(alertController, animated: true)
+    }
+    
+    @IBAction func runBarButtonItemTapped(_ sender: UIBarButtonItem) {
+        guard !phaseSummaries.isEmpty,
+              let expression = initialExpression,
+              let phasesNetworkAttribute = phasesNetworkAttribute,
+              let traceParameters = traceParameters else { return }
+        
+        let traceGroup = DispatchGroup()
+        SVProgressHUD.show(withStatus: "Creating load report…")
+        
+        for i in 0..<phaseSummaries.count {
+            let phase = phaseSummaries[i].phase
+            guard let phaseCode = phase.code else { continue }
+            // Create a conditional expression.
+            let phasesAttributeComparison = AGSUtilityNetworkAttributeComparison(networkAttribute: phasesNetworkAttribute, comparisonOperator: .doesNotIncludeAny, value: phaseCode)!
+            // Chain it with the base condition using an OR operator.
+            traceParameters.traceConfiguration?.traversability?.barriers = AGSUtilityTraceOrCondition(leftExpression: expression, rightExpression: phasesAttributeComparison)
+            
+            traceGroup.enter()
+            utilityNetwork.trace(with: traceParameters) { [weak self] results, _ in
+                defer {
+                    traceGroup.leave()
+                }
+                // Return if not result and ignore any trace error.
+                guard let results = results else { return }
+                var totalCustomers = 0
+                var totalLoad = 0
+                results.forEach { result in
+                    if let elementResult = result as? AGSUtilityElementTraceResult {
+                        // Get the unique customers count.
+                        totalCustomers = Set(elementResult.elements.map { $0.objectID }).count
+                    } else if let functionResult = result as? AGSUtilityFunctionTraceResult {
+                        // Get the total load with a function output.
+                        totalLoad = functionResult.functionOutputs.first?.result as? Int ?? 0
+                    }
+                }
+                self?.phaseSummaries[i].summary = PhaseSummary(phase: phase, totalCustomers: totalCustomers, totalLoad: totalLoad)
```

Comment:
If `self` is no longer around, there's no use in performing any work at all. Therefore I would suggest unwrapping `self` at the beginning of the closure.